internal/hcp/registry/json.go

```diff
@@ -83,7 +83,14 @@ func (h *JSONRegistry) PopulateVersion(ctx context.Context) error {
 
 // StartBuild is invoked when one build for the configuration is starting to be processed
 func (h *JSONRegistry) StartBuild(ctx context.Context, build sdkpacker.Build) error {
-	return h.bucket.startBuild(ctx, build.Name())
+	name := build.Name()
+
+	metadata := build.(*packer.CoreBuild).GetMetadata()
+	err := h.bucket.AddMetadataToBuild(ctx, name, metadata)
```

Comment:
I think `AddMetadataToBuild` belongs to `Version` other than `Bucket`. Version has other similar methods such as `AddLabelsToBuild` and `AddArtifactToBuild` and `AddMetadataToBuild` is breaking the pattern. Was there a specific reason to add it to the bucket?